packages/SwingSet/src/liveslots/stop-vat.js

```diff
@@ -41,7 +41,7 @@ function rejectAllPromises({ m, deciderVPIDs, syscall }) {
   // and attached a .then to it), and stopVat() must not allow
   // userspace to gain agency.
 
-  const rejectCapData = m.serialize('vat upgraded');
+  const rejectCapData = m.serialize(Error('vat upgraded'));
```

Comment:
I was getting complaints from eslint, so assumed it was a mistake.

Reply:
We should definitely be consistent (among all places that announce vat-upgrade). I think I've been slightly conditioned to not use Errors because the `marshal` serialization is so.. enthusiastic.. about announcing them to the console (multiple times), and `stop-vat.js` is new so I might have used a string while writing (and running) the tests.

We do need Notifier subscribers to be able to distinguish between vat-upgrade (meaning they should try again) and vat-failure (meaning they should stop). We could either use string-vs-Error for that, or have clients distinguish between `err.message` (which makes us more dependent upon the actual strings, and reduces our wiggle room for changing them in the future).

@erights as keeper of Notifier, do you think we should make vat-upgrade reject promises with a string, so you can change the Notifier client-side pattern to re-try unless it gets rejected with an Error? Or is there some different/better pattern we should use?

Reply:
I'll open an issue. I feel uncomfortable discriminating based on string vs error instance like this for notifiers

Reply:
@erights @warner @gibson042 can you help me triage https://github.com/Agoric/agoric-sdk/issues/5185
Not sure which label it should fit under.